runtime/drivers/clickhouse/olap.go

```diff
@@ -251,7 +252,86 @@ func (c *connection) InsertTableAsSelect(ctx context.Context, name, sql string,
 		})
 	}
 
-	// merge strategy is also not supported for clickhouse
+	if strategy == drivers.IncrementalStrategyMerge {
+		_, onCluster, err := informationSchema{c: c}.entityType(ctx, "", name)
+		if err != nil {
+			return err
+		}
+		onClusterClause := ""
+		if onCluster {
+			onClusterClause = "ON CLUSTER " + safeSQLName(c.config.Cluster)
+		}
+		// Get the engine info of the given table
+		var engine, engineFull string
+		res, err := c.Execute(ctx, &drivers.Statement{
+			Query:    "SELECT engine, engine_full FROM system.tables WHERE database = currentDatabase() AND name = ?",
+			Args:     []any{name},
+			Priority: 1,
+		})
+		if err != nil {
+			return err
+		}
+		defer res.Close()
+		if res.Next() {
+			if err := res.Scan(&engine, &engineFull); err != nil {
+				return err
+			}
+		}
+		// Distributed table cannot be altered directly, so we need to alter the underlying table
+		if engine == "Distributed" {
```

Comment:
As of now the local table name is fixed : https://github.com/rilldata/rill/blob/b9e61923dd5b6fd074a0b370ae0c153b7dccbc07/runtime/drivers/clickhouse/olap.go#L1017-L1018